src/lib/client.ts

```diff
@@ -852,13 +856,17 @@ export default class MqttClient extends TypedEventEmitter<MqttClientEventCallbac
 			protocolId: this.options.protocolId,
 			protocolVersion: this.options.protocolVersion,
 			clean: this.options.clean,
-			clientId: this.options.clientId,
 			keepalive: this.options.keepalive,
 			username: this.options.username,
-			password: this.options.password as Buffer,
+			// @ts-expect-error - @TODO: @robertsLando this is a dangerous situation as this.options.password can never be a Buffer but was forced as a Buffer
+			password: this.options.password,
 			properties: this.options.properties,
 		}
 
+		if (this.options.clientId !== undefined) {
+			connectPacket.clientId = this.options.clientId
+		}
```

Comment:
clientId should never be undefined, we generate a random one in case it is

Reply:
I agree, but in that case, the corresponding interface `IClientOptions` should reflect that and forbid the property from being `undefined`.